buildSrc/src/main/java/org/elasticsearch/gradle/testfixtures/TestFixturesPlugin.java

```diff
@@ -54,7 +57,16 @@ public void apply(Project project) {
             disableTaskByType(tasks, ThirdPartyAuditTask.class);
             disableTaskByType(tasks, JarHellTask.class);
 
+            Task buildFixture = project.getTasks().create("buildFixture");
+            Task preProcessFixture = project.getTasks().create("preProcessFixture");
+            buildFixture.dependsOn(preProcessFixture);
+            Task postProcessFixture = project.getTasks().create("postProcessFixture");
+            buildFixture.dependsOn(postProcessFixture);
+
             if (dockerComposeSupported(project) == false) {
+                preProcessFixture.setEnabled(false);
```

Comment:
Why even create these if it's not supported?

Reply:
The tasks need to exist because the build script might customize them ( like add a doLast ) . Since that runs at configuration time it would fail when docker-compose is not supported if the tasks would not be created.